symfony/client/src/orangehrmPerformancePlugin/pages/KpiEdit.vue

```diff
@@ -0,0 +1,233 @@
+          <required-text />
+          <oxd-button
+            type="button"
+            display-type="ghost"
+            label="Cancel"
+            @click="onCancel"
+          />
+          <submit-button />
+        </oxd-form-actions>
+      </oxd-form>
+    </div>
+  </div>
+</template>
+
+<script>
+import SwitchInput from '@ohrm/oxd/core/components/Input/SwitchInput';
+import {navigate} from '@ohrm/core/util/helper/navigation';
+import {
+  required,
+  digitsOnly,
+  shouldNotExceedCharLength,
+  minValueShouldBeLowerThanMaxValue,
+} from '@ohrm/core/util/validation/rules';
+import {APIService} from '@/core/util/services/api.service';
+import JobtitleDropdown from '@/orangehrmPerformancePlugin/components/JobtitleDropdown.vue';
+
+const initialKpi = {
+  title: '',
+  jobTitle: null,
+  minRate: null,
+  maxRate: null,
+  isDefault: false,
+};
+
+const rateError = v => {
+  if (digitsOnly(v)) {
+    return (
+      (v >= 0 && v <= 100 && Math.round(v) === v) ||
+      'Should be a number between 0-100'
+    );
+  }
+  return 'Should be a number between 0-100';
+};
+
+export default {
+  name: 'KpiSave',
+  components: {
+    'oxd-switch-input': SwitchInput,
+    'jobtitle-dropdown': JobtitleDropdown,
+  },
+  props: {
+    kpiId: {
+      type: String,
+      required: true,
+    },
+  },
+  setup() {
+    const http = new APIService(
+      'https://02594277-e771-4db3-a5ec-ad645ec49c02.mock.pstmn.io',
+      'api/v2/performance/kpi',
+    );
+
+    return {
+      http,
+    };
+  },
+  data() {
+    return {
+      isLoading: false,
+      kpi: {...initialKpi},
+      rules: {
+        title: [required, shouldNotExceedCharLength(100)],
+        jobTitle: [required],
+        minRate: [
+          required,
+          rateError,
+          v =>
+            !this.kpi.maxRate ||
+            v <= this.kpi.maxRate ||
+            'Minimum Rating should be less than Maximum Rating',
+        ],
+        maxRate: [
+          required,
+          rateError,
+          minValueShouldBeLowerThanMaxValue(
+            () => this.kpi.minRate,
+            'Maximum Rating should be greater than Minimum Rating',
+          ),
+        ],
+      },
+    };
+  },
+  created() {
+    this.isLoading = true;
+    this.http
+      .get(this.kpiId)
+      .then(response => {
+        const {data} = response.data;
+        this.kpi.title = data.title;
+        this.kpi.jobTitle = {id: data.jobTitle.id, label: data.jobTitle.title};
+        this.kpi.minRate = data.minRate;
+        this.kpi.maxRate = data.maxRate;
+        this.kpi.isDefault = data.isDefault;
+      })
+      .finally(() => {
+        this.isLoading = false;
+      });
+  },
+  methods: {
+    onCancel() {
+      navigate('/performance/searchKpi');
+    },
+    onSave() {
+      this.isLoading = true;
+      const jobTitleObj = {
+        id: this.kpi.jobTitle.id,
+        title: this.kpi.jobTitle.label,
+        isDeleted: false,
+      };
+      this.http
+        .create({
```

Comment:
Job title edit should go as `PUT` request. use `this.http.update(id, {data})`

Reply:
Changed it